### ISSUE TITLE
Fallback to correct language when language+region is weird fails in `toRelative()`

### ISSUE THREAD

**Issue by 0xMarK** [malcommac/SwiftDate#635]
I'm using iOS 12.1 simulator (also checked on device). On app start I have:
`Locale.current.languageCode == "fr"`
`Locale.current.regionCode == "US"`
`Locale.preferredLanguages == ["fr-CA", "uk-US", "en-GB", "en"]`

My app supports "en", "fr", "it", etc. So according to [stackoverflow](https://stackoverflow.com/questions/46205763/nsdateformatter-and-current-language-in-ios11/46206511#46206511):

> Under iOS 11, [NSLocale currentLocale] only returns languages supported by your app's localizations.

My `Locale.current.languageCode` is equal `"fr"` (because no "fr-CA" support).

**In SwiftDate `func language(forLocale locale: Locale)` returns incorrect `RelativeFormatterLang`.**

On input I have `locale.collatorIdentifier == "fr_US"` (because `"US"` is my `regionCode`).

The code tries to break the string `localeId.components(separatedBy: "-")` and fails. After that returns _English_...

Expected result:
`func language(forLocale locale: Locale)` returns _French_.

**Comment by 0xMarK** [comment]
Hi @malcommac,

I've made some experiments. It seems that the problem is that I use custom Locale created from currect locale, but it has different `collatorIdentifier`.

For example, if I select language _French_ and region _United States_:

```swift
let currentLocale = Locale.current // fr_US (current)
let currentLocaleCollatorIdentifier = currentLocale.collatorIdentifier // Optional("fr")

let customLocale = Locale(identifier: Locale.current.identifier) // fr_US (fixed)
let localeCollatorIdentifier = customLocale.collatorIdentifier // Optional("fr_US")
```

But if I select language _French (Canada)_ and region _United States_:

```swift
let currentLocale = Locale.current // fr-CA_US (current)
let currentLocaleCollatorIdentifier = currentLocale.collatorIdentifier // Optional("fr-CA")

let customLocale = Locale(identifier: Locale.current.identifier) // fr-CA_US (fixed)
let localeCollatorIdentifier = customLocale.collatorIdentifier // Optional("fr-CA_US")
```

The problem is that in the situation when customLocale.collatorIdentifier is equal to "fr_US" – `func language(forLocale locale: Locale)` falls back to **English**.

**Comment by xmanu** [comment]
I can confirm this issue for a standard German iPhone setup.

Calling `until.toRelative(since: from, style: RelativeFormatter.defaultStyle(), locale: Locale.current)` will fallback to english, as `locale.collatorIdentifier` returns `de-DE`. This seems undesirable. I propose adding another guard fallback to only use the languageCode:

`
guard let table = self.languagesCache[localeId] else {
  guard let tableType = self.languagesMap[localeId] ?? self.languagesMap[locale.languageCode ?? "en"] else {
    return language(forLocale: Locales.english.toLocale())
  }
...
`

**Comment by malcommac** [comment]
Would you make a PR for this?

**Comment by xmanu** [comment]
Sure. I’ll get to it tomorrow...

**Comment by xmanu** [comment]
I have created a pull request, that to the best of my knowledge, fix the issue at hand. I had to adapt my fix to also cover the original authors issue.

I was only able to verify this issue by manually creating a fr_US locale, so I would be glad to see this verified by the author of this issue.

The proposed PR should not be breaking any existing cases though, only return potentially more precise language tables.
